src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -181,21 +181,26 @@ QtObject:
     self.errorCodeChanged()
 
   proc processResponse(self: Controller, response: JsonNode) =
-    defer: self.setLoadingData(false)
-
     let res = fromJson(response, backend_activity.FilterResponse)
 
     defer: self.setErrorCode(res.errorCode.int)
 
+    if res.errorCode == ErrorCodeFilterCanceled and self.model.getCount() == 0:
```

Comment:
Still trying to figure this out. It looks like hiding errors and no results under the loading state. This might confuse users, thinking that something is working. Why not show an error state in case of error and empty state in case of no data?

Reply:
Indeed this solves the multiple answers part, but it is a workaround. A proper fix should be something similar to what I did with the other async calls in my PR here https://github.com/status-im/status-desktop/blob/5397b4705ceb78ef97d228d847451239b653db0a/src/app/modules/main/wallet_section/activity/controller.nim#L52C6-L52C6; sorry that I missed the activity fetching requires the same behaviour, will update the PR with this fix.

Ideally, we should only have two calls if required. It is required when invalidating the filtering data, which probably happens here due to the UI's reactive nature.